### ISSUE TITLE
[BUG]TGLU_RVP_NOCODEC_CI ipc error when issueing GLB_TPLG_MSG:COMP_CONNECT 

### ISSUE THREAD

**Issue by XiaoyunWu6666** [thesofproject/sof#4684]
**Describe the bug**
found in inner daily 6090
after firmware boot complete.
TGLU_RVP_NOCODEC_CI ipc error when issueing GLB_TPLG_MSG:COMP_CONNECT while  setting up route MUXDEMUX1.0 -> BUF7.0

**To Reproduce**
TPLG=/lib/firmware/intel/sof-tplg/sof-tgl-nocodec-ci.tplg ~/sof-test/test-case/check-signal-stop-start.sh -m playback -c 50

happened first in inner daily 6090 and secondly in 6200

**Environment**
Kernel Branch: topic/sof-dev
Kernel Commit: eb558167
SOF Branch: main
SOF Commit: 73803adec979
Topology:  /lib/firmware/intel/sof-tplg/sof-tgl-nocodec-ci.tplg
![image](https://user-images.githubusercontent.com/79959904/131058975-905a035a-7fca-406b-b974-68042be1e93b.png)

 Platform: TGLU_RVP_NOCODEC_CI

**Screenshots or console output**
[console]
```
2021-08-25 22:06:37 UTC [REMOTE_INFO] Testing: run stop/start test on PCM:smart-nocodec,playback. Interval time: 0.5
2021-08-25 22:06:37 UTC [REMOTE_COMMAND] aplay -Dhw:0,2 -r 48000 -c 2 -f S16_LE /dev/zero -q &
aplay: main:830: audio open error: Invalid argument
lsof: WARNING: can't stat() fuse.gvfsd-fuse file system /run/user/125/gvfs
      Output information may be incomplete.
2021-08-25 22:06:37 UTC [REMOTE_ERROR] aplay process[16447] is terminated too early
```
[dmesg]
```
[ 2634.632629] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: setting up route MUXDEMUX6.0 -> BUF6.0
[ 2634.632634] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx: 0x30030000: GLB_TPLG_MSG: COMP_CONNECT
[ 2634.632794] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx succeeded: 0x30030000: GLB_TPLG_MSG: COMP_CONNECT
[ 2634.632798] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: setting up route BUF6.0 -> PCM2C
[ 2634.632803] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx: 0x30030000: GLB_TPLG_MSG: COMP_CONNECT
[ 2634.632963] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx succeeded: 0x30030000: GLB_TPLG_MSG: COMP_CONNECT
[ 2634.632969] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: setting up route MUXDEMUX1.0 -> BUF7.0
[ 2634.632974] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx: 0x30030000: GLB_TPLG_MSG: COMP_CONNECT
[ 2634.633081] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx error for 0x30030000 (msg/reply size: 16/12): -22
[ 2634.633091] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: sof_route_setup_ipc: route setup failed -22
[ 2634.633095] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: sof_set_up_pipelines: restore pipeline connections failed
[ 2634.633099] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: failed to restore pipeline after resume -22
[ 2634.633108] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ASoC: error at snd_soc_pcm_component_pm_runtime_get on 0000:00:1f.3: -22
[ 2634.633117] kernel:  NoCodec-2: soc_pcm_open() failed (-22)
[ 2634.633122] kernel:  smart-nocodec: ASoC: dpcm_be_dai_startup() failed at NoCodec-2 (-22)
[ 2634.633128] kernel:  smart-nocodec: dpcm_fe_dai_startup() failed (-22)
[ 2635.130251] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[ 2635.130290] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[ 2635.338380] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: Copying cached debugfs data
[ 2635.338428] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: Copying cached debugfs data
[ 2635.339059] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: Copying cached debugfs data
[ 2635.339065] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: Copying cached debugfs data
[ 2635.630192] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[ 2635.630245] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[ 2636.573495] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: enabling device failed: -22
[ 2636.573501] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: fail in sof_ipc_trace_update_filter -22
```

**Comment by XiaoyunWu6666** [comment]
in inner daily 6200 , on model TGLU_RVP_NOCODEC_CI , in case check-suspend-resume-with-playback-5
ipc timed out when COMP_FREE and brought DSP panic when freeing widget PCM2P

**Comment by XiaoyunWu6666** [comment]
in inner daily 6200 , on model TGLU_RVP_NOCODEC_CI , in case check-suspend-resume-with-playback-5
ipc timed out when COMP_FREE and brought DSP panic when freeing widget PCM2P
[dmesg]
```
[ 3013.270563] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx: 0x30020000: GLB_TPLG_MSG: COMP_FREE
[ 3013.770619] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[ 3013.770661] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[ 3013.774003] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx timed out for 0x30020000 (msg/reply size: 12/12)
[ 3013.774014] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: preventing DSP entering D3 state to preserve context
[ 3013.774018] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ------------[ IPC dump start ]------------
[ 3013.774032] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: hda irq intsts 0x00000000 intlctl 0xc0000001 rirb 00
[ 3013.774037] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: dsp irq ppsts 0x00000000 adspis 0x00000000
[ 3013.774044] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: host status 0x00000000 dsp status 0x00000000 mask 0x00000003
[ 3013.774048] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ------------[ IPC dump end ]------------
[ 3013.774051] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ------------[ DSP dump start ]------------
[ 3013.774057] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: status: fw entered - code 00000005
[ 3013.774431] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: unexpected fault 0x00000000 trace 0x00004000
[ 3013.774435] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ------------[ DSP dump end ]------------
[ 3013.774440] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: failed to free widget PCM2P
[ 3013.774450] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx: 0x40070000: GLB_PM_MSG: CORE_ENABLE
[ 3014.270586] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG
[ 3014.270637] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG
[ 3014.277985] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx timed out for 0x40070000 (msg/reply size: 12/12)
[ 3014.277999] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: failed to disable target core: 1 for widget PCM2P
[ 3014.278016] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: FW Poll Status: reg[0x80]=0x20140000 successful
[ 3014.278052] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx: 0x40010000: GLB_PM_MSG: CTX_SAVE
[ 3014.781938] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx timed out for 0x40010000 (msg/reply size: 76/12)
[ 3014.781950] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ctx_save ipc error -110, proceeding with suspend
```
[dmesg.txt](https://github.com/thesofproject/sof/files/7081418/dmesg.txt)
[sof-logger]
```
[       27689.061400] (         187.708328) c0 ipc                   src/ipc/ipc3/helper.c:585  ERROR ipc_comp_connect(): source component does not exist, source_id = 2 sink_id = 40
```

**Comment by XiaoyunWu6666** [comment]
@keyonjie 
This may be duplicate of #4552 , since many testcases 'have to' make DUTs runtime resume before tests begin ,  runtime suspend/resume errors tend to spread .

**Comment by keyonjie** [comment]
@juimonen @lyakh there are several related issues filed, they happens during runtime resuming stage, and looks they are multi-core related, and the error can happen quite randomly with different IPCs.

**Comment by lgirdwood** [comment]
@XiaoyunWu6666 I've taken your latest dmesg and can see we have a NULL ptr dereference after the D0 entry (please make sure you get the FW crash dumps in the log if found). 
1) We enter D3 and get the IPC timout (in your log reply above)
2) We enter D0 and crash on stream msg
```
[ 3019.667759] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx succeeded: 0x80010000: GLB_DAI_MSG: CONFIG
[ 3019.667774] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: pcm: prepare stream 2 dir 0
[ 3019.667776] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: pcm: hw params stream 2 dir 0
[ 3019.667777] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: generating page table for 00000000f6a8f163 size 0x10000 pages 16
[ 3019.667790] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: FW Poll Status: reg[0x160]=0x40000 successful
[ 3019.667832] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: FW Poll Status: reg[0x160]=0x40000 successful
[ 3019.667834] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: hda_dsp_stream_setup_bdl: period_bytes:0x4000
[ 3019.667836] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: hda_dsp_stream_setup_bdl: periods:4
[ 3019.667855] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: stream_tag 1
[ 3019.667864] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: tplg: complete pipeline PIPELINE.5.SSP2.OUT id 32
[ 3019.667867] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx: 0x30130000: GLB_TPLG_MSG: PIPE_COMPLETE
[ 3019.668582] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx succeeded: 0x30130000: GLB_TPLG_MSG: PIPE_COMPLETE
[ 3019.668602] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx: 0x60010000: GLB_STREAM_MSG: PCM_PARAMS
<span class="txtRed">[ 3019.668918] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error : DSP panic!</span>
<span class="txtRed">[ 3019.668930] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: panic: dsp_oops_offset 788480 offset 788480</span>
[ 3019.668932] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ------------[ DSP dump start ]------------
[ 3019.668945] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: status: fw entered - code 00000005
<span class="txtRed">[ 3019.669048] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: runtime exception</span>
<span class="txtRed">[ 3019.669057] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: trace point 00004000</span>
<span class="txtRed">[ 3019.669067] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: panic at :0</span>
<span class="txtRed">[ 3019.669075] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: DSP Firmware Oops</span>
<span class="txtRed">[ 3019.669085] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: Exception Cause: LoadStorePIFDataErrorCause, Synchronous PIF data error during LoadStore access</span>
[ 3019.669100] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: EXCCAUSE 0x0000000d EXCVADDR 0x00000040 PS       0x00060025 SAR     0x00000000
[ 3019.669115] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: EPC1     0xbe045be5 EPC2     0xbe0467f2 EPC3     0x00000000 EPC4    0x00000000
[ 3019.669129] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: EPC5     0x00000000 EPC6     0x00000000 EPC7     0x00000000 DEPC    0x00000000
[ 3019.669143] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: EPS2     0x00060920 EPS3     0x00000000 EPS4     0x00000000 EPS5    0x00000000
[ 3019.669156] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: EPS6     0x00000000 EPS7     0x00000000 INTENABL 0x00000000 INTERRU 0x00000222
[ 3019.669170] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: stack dump from 0xbe119900
[ 3019.669179] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: 0xbe119900: 00000004 00000001 00000002 9e1fd600
[ 3019.669190] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: 0xbe119904: 7e045be5 be119980 9e19e140 be090000
[ 3019.669202] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: 0xbe119908: a55e6200 fa20a28f 71215028 ffff9c09
[ 3019.669213] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: 0xbe11990c: 000c0800 00000000 71215028 ffff9c09
[ 3019.669225] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: 0xbe119910: 71215028 ffff9c09 00000092 00000000
[ 3019.669236] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: 0xbe119914: 809f4400 ffff9c09 83fc0400 ffff9c09
[ 3019.669247] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: 0xbe119918: a55e6200 fa20a28f 91acffa0 ffffffff
[ 3019.669256] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: 0xbe11991c: 8093ca80 ffff9c09 91a8d710 ffffffff
[ 3019.669264] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ------------[ DSP dump end ]------------
```

**Comment by keyonjie** [comment]
should already be fixed with main or stable-v2.0

**Comment by lgirdwood** [comment]
> should already be fixed with main or stable-v2.0

Ack - not seen this in daily for a long time especially after the D3 fixes merged - we can close now and reopen if it comes back.
